src/pages/Game.tsx

```diff
@@ -146,15 +146,72 @@ const GamePage = ({ ws, currentUser }: GameProps): JSX.Element => {
    */
   const gameStartInit = (payload: unknown): void => {
     const data = payload as GameStartJSON;
-    const tileArray = data.tiles;
-    const tiles: Tile[] = [];
-    tileArray.forEach((tile: string) => {
-      tiles.push(TileFactory.createTileFromStringDef(tile));
-    });
+    const { tiles: tileArray, selfPlayedTiles, currentIndex } = data;
+
     const mjGameState = gameState as MahjongGameState;
 
-    const mjPlayer = mjGameState.getMjPlayer();
+    // Increment wall counter
+    const wallCounter = mjGameState.getWallCounter();
+    const result = wallCounter.setCurrentIndex(currentIndex);
+    if (!result) console.error('Failed to set wall counter. Please verify game state');
+
+    // Set player hand
+    const tiles = tileArray.map((tile: string) => TileFactory.createTileFromStringDef(tile));
+    const mjPlayer = mjGameState.getMjPlayer() as MahjongPlayer;
     mjPlayer.setHand(tiles);
+    const mjPlayerHand = mjPlayer.getHand();
+
+    // Set player played tiles (bonus tiles) at game start
+    const playerPlayedTileObj = selfPlayedTiles.find(
+      (playedTiles: SelfPlayedTile) => playedTiles.connectionId === mjPlayer.getConnectionId(),
+    ) as SelfPlayedTile;
+    const mjPlayerBonusTilesStrDef = playerPlayedTileObj.playedTiles;
+
+    if (mjPlayerBonusTilesStrDef.length !== 0) {
+      const mjPlayerBonusTiles = mjPlayerBonusTilesStrDef.map((strDef) => TileFactory.createTileFromStringDef(strDef));
+      mjPlayerHand.addPlayedTiles(mjPlayerBonusTiles);
+    }
+    console.log(
+      `Game - gameStartInit():\n Player '${mjPlayer.getName()}', playerBonusTiles: [${mjPlayerBonusTilesStrDef}]`,
+    );
+
+    // Get opponents in mjGameState
+    const mjOpponents = mjGameState
+      .getUsers()
+      .filter((user) => user.getConnectionId() !== mjPlayer.getConnectionId()) as MahjongOpponent[];
+    const opponentsBonusTilesObjs = selfPlayedTiles.filter(
+      (playedTiles) => playedTiles.connectionId !== mjPlayer.getConnectionId(),
+    );
+
+    // Set opponents played tiles (bonus tiles) at game start
+    let errorAddingBonusTiles = false;
+    const failedOpponents: MahjongOpponent[] = [];
+    mjOpponents.forEach((opponent) => {
+      const opponentHand = opponent.getHand();
+      const opponentPlayedTileObj = opponentsBonusTilesObjs.find(
+        (obj) => obj.connectionId === opponent.getConnectionId(),
+      );
+      if (opponentPlayedTileObj) {
+        const opponentPlayedTilesStrDef = opponentPlayedTileObj.playedTiles;
+        // eslint-disable-next-line
+        const opponentBonusTiles = opponentPlayedTilesStrDef.map((tile) => {
+          return TileFactory.createTileFromStringDef(tile);
+        });
```

Comment:
I'm assuming this violates line length ?
```suggestion
        const opponentBonusTiles = opponentPlayedTilesStrDef.map((tile) => TileFactory.createTileFromStringDef(tile));
```

Reply:
can change to that, sometimes eslint is conflicting with prettier